checkout/access-checkout-android/src/main/java/com/worldpay/access/checkout/views/SessionResponseListener.kt

```diff
@@ -2,8 +2,22 @@ package com.worldpay.access.checkout.views
 
 import com.worldpay.access.checkout.api.AccessCheckoutException
 
+/**
+ * The [SessionResponseListener] will receive an update once the session state has been generated. The listener will also
+ * be notified of any errors that could have occurred during the request.
+ */
 interface SessionResponseListener {
+
+    /**
+     * Method for being notified when the request for the session state starts
+     */
     fun onRequestStarted()
-    fun onRequestFinished(sessionReference: String?, error: AccessCheckoutException?)
+
+    /**
+     * Method for being notified when the session state is available, or an error has occurred when trying to generate one
+     * @param (Optional) sessionState when the session has been created successfully
+     * @param (Optional) [AccessCheckoutException] when there has been an error generating the session state
```

Comment:
@param error (Optional) a subclass of [AccessCheckoutException] when there has been an error generating the session state